### PR TITLE
BoS and Vault nets separated again

### PR DESCRIPTION
**Pull request by nokele**
vault gets the old 2.1k research/minute, BoS keeps 210.

### CODE REVIEW

File: code/controllers/subsystem/research.dm [hunk 1]
```diff
@@ -351,22 +357,30 @@ SUBSYSTEM_DEF(research)
 				result[i] *= eff
 				bitcoins[i] = bitcoins[i]? bitcoins[i] + result[i] : result[i]
 	else
-		for(var/obj/machinery/rnd/server/miner in servers)
+		for(var/obj/machinery/rnd/server/miner in BOSservers)
+			if(miner.working)
+				BOSbitcoins = BOSsingle_server_income.Copy()
+				//break
```
Comment (madeleine-lavenza) [inline code/controllers/subsystem/research.dm:363]: Why was this break removed? It literally just wastes processing time to remove it.

Comment (MarinaGryphon) [inline code/controllers/subsystem/research.dm:363]:
```suggestion
				break
```
Why was this break removed? It literally just wastes processing time to remove it.

File: code/controllers/subsystem/research.dm [hunk 2]
```diff
@@ -351,22 +357,30 @@ SUBSYSTEM_DEF(research)
...
+		for(var/obj/machinery/rnd/server/miner in VAULTservers)
 			if(miner.working)
-				bitcoins = single_server_income.Copy()
-				break			//Just need one to work.
+				VAULTbitcoins = VAULTsingle_server_income.Copy()
+				//break
```
Comment (MarinaGryphon) [inline code/controllers/subsystem/research.dm:367]:
```suggestion
				break	
```
Why was this break removed? It literally just wastes processing time to remove it.

File: code/controllers/subsystem/research.dm [hunk 3]
```diff
@@ -294,7 +297,8 @@ SUBSYSTEM_DEF(research)
 	var/list/errored_datums = list()
 	var/list/point_types = list()				//typecache style type = TRUE list
 	//----------------------------------------------
-	var/list/single_server_income = list(TECHWEB_POINT_TYPE_GENERIC = 3.5)	//citadel edit - techwebs nerf
+	var/list/BOSsingle_server_income = list(TECHWEB_POINT_TYPE_GENERIC = 3.5)	//citadel edit - techwebs nerf
+	var/list/VAULTsingle_server_income = list(TECHWEB_POINT_TYPE_GENERIC = 35)
```
Comment (MarinaGryphon) [inline code/controllers/subsystem/research.dm:301]: The vault has 10x the research of the BoS? Yeah, right...

File: code/modules/research/server.dm [hunk 4]
```diff
@@ -166,21 +166,21 @@
...
-	SSresearch.servers |= src
+	SSresearch.VAULTservers |= src
 	stored_research = SSresearch.science_tech //note this lettern
 	var/obj/item/circuitboard/machine/B = new /obj/item/circuitboard/machine/rdserver(null)
 	B.apply_default_parts(src)
 
 /obj/machinery/rnd/server/followers/Initialize()
 	. = ..()
-	SSresearch.servers |= src
+	SSresearch.VAULTservers |= src
```
Comment (MarinaGryphon) [inline code/modules/research/server.dm:183]: The Followers now use the Vault's network?

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]
This is a trainwreck. Genuinely. I would advise a ground-up rewrite but that's not going to happen, since Corvo just hits merge on anything that passes CI regardless of if it's even finished, much less functional, so in that case this review just addresses a few obvious concerns.

### If you want to actually fix it:
I would suggest making the servers list keyed by an ID var on RND servers. Something like `var/list/list/obj/machinery/rnd/server/servers = list()`, and then something like `LAZYDISTINCTADD(SSresearch.servers, SERVER_ID_VAULT, src)`/`LAZYDISTINCTADD(SSresearch.servers, SERVER_ID_BOS, src)`, etc. Then instead of duplicating the for loop in `fire()`, you can just do
```
for(var/server_id in servers)
    for(var/obj/machinery/rnd/server/miner in servers[server_id])
        LAZYSET(bitcoins, server_id, single_server_income[server_id].Copy())
```
This would allow you to have arbitrarily many servers. You'd also have to modify the `multiserver_calculation` bit, and change `calculate_server_coefficient()` to `calculate_server_coefficient(server_id)`, + modify that code to use `servers[id].len` instead of `servers.len`, etc.

I would also advise to actually understand the code you're modifying before you do damage to the server's subsystems. This kind of blind edit is how Bad Deathclaw et al. got to be so bad.

File: code/modules/research/server.dm (hunk 4 above)
Comment (nokele) [reply code/modules/research/server.dm:183]: followers don't have research anymore.

File: code/controllers/subsystem/research.dm (hunk 3 above)
Comment (nokele) [reply code/controllers/subsystem/research.dm:301]: BoS has other ways to get research now.

File: code/controllers/subsystem/research.dm (hunk 1 above)
Comment (nokele) [reply code/controllers/subsystem/research.dm:363]: will take a look at the rest.

Comment (nokele) [reply code/controllers/subsystem/research.dm:363]: Ok honestly no idea what I'm doing apart from removing the breaks. Tried for a bit and failed. Not a CS student/worker here as you can see so you'll have to explain this in more detail if you want me to actually manage to do anything.

Comment (judgex) [reply code/controllers/subsystem/research.dm]: Oh shit the CI finished?  *slaps merge without looking*  Yeeeeeeeeeeeeeeeeeeeeeeeeee haawwwwwwww

**Pull request merged**